File: app/src/main/java/com/android_cademy/starwarstinder/ViewModel/MainViewModelFactory.java
package com.android_cademy.starwarstinder.ViewModel;

import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import com.android_cademy.starwarstinder.model.ProfileRepository;

public class MainViewModelFactory extends ViewModelProvider.NewInstanceFactory {

    private final ProfileRepository mRepository;

    public MainViewModelFactory(ProfileRepository repository) {
        this.mRepository = repository;
    }

    @Override
    public <T extends ViewModel> T create(Class<T> modelClass) {
        //noinspection unchecked
        return (T) new MainViewModel(mRepository);
    }
}

File: app/src/main/java/com/android_cademy/starwarstinder/ViewModel/MainViewModelFactory.kt
package com.android_cademy.starwarstinder.ViewModel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

import com.android_cademy.starwarstinder.model.ProfileRepository

class MainViewModelFactory(private val mRepository: ProfileRepository) : ViewModelProvider.NewInstanceFactory() {

    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return MainViewModel(mRepository) as T
    }

    //    @Override
    //    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
    //
    //    }
}